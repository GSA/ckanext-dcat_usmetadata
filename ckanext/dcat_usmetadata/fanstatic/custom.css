/* Hide original "Add dataset" button */
.page_primary_action > a.btn-primary:nth-of-type(2) {
    display: none !important;
}
