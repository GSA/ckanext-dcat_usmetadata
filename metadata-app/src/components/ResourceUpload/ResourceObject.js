export default {
  name: '',
  description: '',
  url: '',
  format: '',
  mimetype: '',
  upload: null,
};
