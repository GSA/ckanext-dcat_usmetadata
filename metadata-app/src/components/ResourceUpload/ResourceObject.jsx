export default {
  name: '',
  description: '',
  url: '',
  format: '',
  mimetype: '',
  conformsTo: '',
  upload: null,
};
