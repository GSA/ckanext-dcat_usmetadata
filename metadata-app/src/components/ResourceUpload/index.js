import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import { Dialog } from '@cmsgov/design-system';
import Link from '../Link';
import Radio from '../Radio';
import WrappedField from '../WrappedField';
import resourceFormats from './resource_formats.json';
import { RESOURCE_URL_TYPES } from '../../api';

import '@cmsgov/design-system/dist/css/index.css';

const sortedResourceFormats = resourceFormats.sort((a, b) => {
  const labelA = a.label.toUpperCase();
  const labelB = b.label.toUpperCase();
  if (labelA < labelB) {
    return -1;
  }
  if (labelA > labelB) {
    return 1;
  }

  return 0;
});

const ResourceUpload = (props) => {
  const {
    values,
    errors,
    setFieldValue,
    submitForm,
    draftSaved,
    isSubmitting,
    handleSteps,
  } = props; // eslint-disable-line
  const { resourceAction } = values;
  const resource = values.resource || {};
  const resources = values.resources || [];
  const { url, name, description, mimetype, format } = resource;

  const [uploadDataFileIsActive, setUploadDataFileActive] = useState(false);
  const [shouldShowModal, setShowModal] = useState(false);

  // Detect when url type changes
  useEffect(() => {
    // If it's selected Link to an API radio option then
    // set format value to API
    if (!resource.format && resource.urlType === RESOURCE_URL_TYPES.LINK_TO_API) {
      setFieldValue('resource.format', 'API');
    }
  }, [resource.urlType]);

  const handleFileChange = (event) => {
    setUploadDataFileActive(!uploadDataFileIsActive);
    setFieldValue('resource.fileName', event.currentTarget.files[0].name);
    if (!name) {
      setFieldValue('resource.name', event.currentTarget.files[0].name);
    }
    if (!mimetype) {
      setFieldValue('resource.mimetype', event.currentTarget.files[0].type);
    }
    if (!format) {
      const detectedFormat =
        event.currentTarget.files[0].name &&
        event.currentTarget.files[0].name.split('.').slice(-1)[0];
      setFieldValue('resource.format', detectedFormat || '');
    }
    setFieldValue('resource.upload', event.currentTarget.files[0]);
  };

  const hideModal = () => {
    setShowModal(false);
  };

  const showModal = (resourceName) => {
    setShowModal(resourceName);
  };

  const editResource = (res) => {
    setFieldValue('resourceAction', 'edit');
    setUploadDataFileActive(false);
    setFieldValue('resource', res);
    setFieldValue('resource.size', '');
    setFieldValue('resource.webstore_last_updated', '');
    setFieldValue('resource.cache_last_updated', '');
  };

  const deleteResource = (res) => {
    setFieldValue('resource', { id: res.id });
    setFieldValue('resourceAction', 'delete');
    setFieldValue('publish', false);
    setFieldValue('saveDraft', false);
    setUploadDataFileActive(false);
    submitForm();
  };

  return (
    <div className="usa-form-custom" id="resource-upload">
      <section id="section-basic-mega-menu" className="site-component-section">
        <h1 className="usite-page-title" id="basic-mega-menu">
          Resource Upload
        </h1>
        <p className="site-text-intro">
          You can add the URL of the dataset where it is available on the agency website. If you are
          uploading the dataset itself, please notify the Data.gov team at{' '}
          <Link target="_blank" href="mailto:inventory-help@gsa.gov">
            inventory-help@gsa.gov
          </Link>
          . You can also add a URL or file of information related to the dataset such as a data
          dictionary.
        </p>
      </section>
      {!resources.length ? (
        ''
      ) : (
        <div className="margin-top-10 padding-bottom-8 border-gray-10 border-bottom-2px">
          <div className="grid-row margin-top-3">
            <div className="grid-col-12 text-custom-green">
              <i>Resource saved: ({resources.length} resources saved in total)</i>
            </div>
          </div>
          {resources.map((res) => {
            return (
              <div key={res.id} className="grid-row margin-top-2">
                <div className="grid-col-12 bg-base-lightest padding-x-1">
                  <div className="resource-item-name">{res.name}</div>
                  <button
                    id={`delete-${res.name}`}
                    onClick={() => showModal(res.name)}
                    type="button"
                    className="usa-button--unstyled resource-action-button"
                  >
                    Delete
                  </button>
                  {shouldShowModal === res.name && (
                    <Dialog
                      onExit={() => hideModal()}
                      getApplicationNode={() => document.getElementById('resource-upload')}
                      heading="Please Confirm Action"
                      underlayClickExits
                      actions={[
                        <button
                          type="button"
                          className="ds-c-button ds-c-button--primary"
                          key="primary"
                          onClick={() => {
                            deleteResource(res);
                            hideModal();
                          }}
                        >
                          Delete
                        </button>,
                        <button
                          type="button"
                          className="ds-c-button ds-c-button--transparent"
                          key="cancel"
                          onClick={() => hideModal()}
                        >
                          Cancel
                        </button>,
                      ]}
                    >
                      Are you sure you want to delete this resource?
                    </Dialog>
                  )}
                  <button
                    id={`edit-${res.name}`}
                    onClick={() => {
                      editResource(res);
                    }}
                    type="button"
                    className="usa-button--unstyled resource-action-button"
                  >
                    Edit
                  </button>
                </div>
              </div>
            );
          })}
        </div>
      )}
      {
        <div className="grid-row margin-top-3">
          <div className="grid-col-12">
            <span className="usa-label">Resource</span>
            <Radio
              label={
                // eslint-disable-next-line react/jsx-wrap-multilines
                <>
                  <b>Link to a file:</b> Provide a direct download link to a file
                </>
              }
              name="resource.urlType"
              value={RESOURCE_URL_TYPES.LINK_TO_FILE}
              id="resource-option-link-to-file"
            />
            <Radio
              label={
                // eslint-disable-next-line react/jsx-wrap-multilines
                <>
                  <b>Upload a file:</b> Upload a file from your computer
                </>
              }
              name="resource.urlType"
              value={RESOURCE_URL_TYPES.UPLOAD_FILE}
              id="resource-option-upload-file"
            />
            <Radio
              label={
                // eslint-disable-next-line react/jsx-wrap-multilines
                <>
                  <b>Link to an API:</b> Provide a link to access a dataset via API
                </>
              }
              name="resource.urlType"
              value={RESOURCE_URL_TYPES.LINK_TO_API}
              id="resource-option-link-to-api"
            />
            <Radio
              label={
                // eslint-disable-next-line react/jsx-wrap-multilines
                <>
                  <b>Access URL:</b> Provide a link to a resource that is not directly downloadable,
                  like an html website
                </>
              }
              name="resource.urlType"
              value={RESOURCE_URL_TYPES.ACCESS_URL}
              id="resource-option-access-url"
            />
          </div>
          <div className="grid-col-12">
            {resource.urlType && (
              // eslint-disable-next-line jsx-a11y/label-has-associated-control
              <label className="usa-label">
                {resource.urlType === RESOURCE_URL_TYPES.UPLOAD_FILE ? 'File' : 'URL'}
              </label>
            )}
            {/* Do not show URL field unless it's url type or url already exists (edit mode) */}
            {/* eslint-disable-next-line */}
            {(resource.urlType && resource.urlType !== RESOURCE_URL_TYPES.UPLOAD_FILE) || url ? (
              <div>
                <p className="usa-helptext">
                  {`If you are linking to a dataset, please include "https://" at the beginning
                of your URL.`}
                </p>
                <WrappedField
                  id="url"
                  name="resource.url"
                  type="url"
                  value={url}
                  onClick={() => {
                    setFieldValue('resource.url', '');
                  }}
                  errors={errors}
                />
              </div>
            ) : uploadDataFileIsActive ? (
              // If the file is uploaded
              <WrappedField
                name="resource.fileName"
                type="label"
                value={resource.fileName}
                onClick={() => {
                  setFieldValue('resource.upload', null);
                  setFieldValue('resource.fileName', '');
                  setFieldValue('resource.name', '');
                  setFieldValue('resource.description', '');
                  setFieldValue('resource.format', '');
                  setFieldValue('resource.mimetype', '');
                  setUploadDataFileActive(false);
                }}
              />
            ) : (
              // If the file is not uploaded and the radio button is chosen local file
              resource.urlType === RESOURCE_URL_TYPES.UPLOAD_FILE && (
                <>
                  <br />
                  {/* eslint-disable-next-line */}
                  <label tabIndex="0" htmlFor="upload" className="usa-button usa-button--base">
                    <i className="fa fa-cloud-upload" aria-hidden="true" /> Upload data
                  </label>
                  <input
                    id="upload"
                    name="resource.upload"
                    type="file"
                    onChange={handleFileChange}
                  />

                  <p className="usa-helptext">
                    Formats accepted include the following: TXT, HTML, TSV, CSV, ODT, XML, Perl.
                  </p>
                </>
              )
            )}
          </div>
        </div>
      }
      <div className="grid-row margin-top-3">
        <div className="grid-col-12">
          <WrappedField
            label="Name"
            name="resource.name"
            type="string"
            placeholder=""
            helptext=""
            value={name}
          />
        </div>
      </div>
      <div className="grid-row margin-top-3">
        <div className="grid-col-12">
          <WrappedField
            label="Description"
            name="resource.description"
            type="string"
            component="textarea"
            rows="6"
            helptext="You can use Markdown Formatting here."
            value={description}
          />
        </div>
      </div>
      <div className="grid-row margin-top-3">
        <div className="grid-col-12">
          <WrappedField
            label="Media Type"
            name="resource.mimetype"
            type="select"
            choices={sortedResourceFormats}
            value={mimetype}
          />
        </div>
      </div>
      <div className="grid-row margin-top-3">
        <div className="grid-col-12">
          <WrappedField
            label="Format"
            name="resource.format"
            type="string"
            helptext="Examples include: csv, xml, json.  This will be guessed automatically.  Leave blank if you wish."
            value={format}
          />
        </div>
      </div>

      <div className="grid-row margin-top-3">
        <div className="grid-col-12">
          <button
            type="button"
            className="usa-button usa-button--outline"
            onClick={() => {
              setFieldValue('publish', false);
              setFieldValue('saveDraft', false);
              setUploadDataFileActive(false);
              submitForm();
            }}
            disabled={isSubmitting}
          >
            {resourceAction === 'edit' ? 'Save' : 'Save and add another resource'}
          </button>
        </div>
      </div>
      {values.lastSavedResource && resources.length ? (
        <div className="grid-row margin-top-3">
          <div className="grid-col-12">
            <div className="usa-alert usa-alert--success usa-alert--slim usa-alert--no-icon">
              <div className="usa-alert__body">
                <p className="usa-alert__text">
                  <i>
                    Resource saved: [{values.lastSavedResource}] ({resources.length} resources saved
                    in total).
                  </i>
                  <br />
                </p>
              </div>
            </div>
          </div>
        </div>
      ) : (
        ''
      )}
      {isSubmitting ? (
        <div className="usa-alert usa-alert--info usa-alert--slim usa-alert--no-icon">
          <div className="usa-alert__body">
            <p className="usa-alert__text">
              Your submission is in progress, this may take a few minutes for large dataset files.
            </p>
          </div>
        </div>
      ) : (
        ''
      )}

      <div className="margin-top-6 clearfix">
        {values.publishing_status === 'Published' ? (
          ''
        ) : (
          <button
            className="usa-button usa-button--outline"
            type="button"
            onClick={async () => {
              await setFieldValue('publish', false);
              await setFieldValue('saveDraft', true);
              submitForm();
            }}
          >
            Save draft
          </button>
        )}
        <div className="float-right">
          <button
            className="usa-button usa-button--outline"
            type="button"
            onClick={() => handleSteps(1)}
            onKeyUp={(e) => {
              if (e.keyCode === 13) {
                handleSteps(1);
              }
            }}
          >
            Back to previous page
          </button>
          <button
            className="usa-button margin-right-0"
            type="button"
            onClick={() => {
              setFieldValue('publish', true);
              submitForm();
            }}
            disabled={isSubmitting}
          >
            Finish and publish
          </button>
        </div>
      </div>

      {draftSaved && (
        <div style={{ marginTop: '1rem' }}>
          <div className="grid-col-12 text-custom-green">
            <i>
              Draft saved:
              <br />[{draftSaved}]
            </i>
          </div>
        </div>
      )}
    </div>
  );
};

ResourceUpload.propTypes = {
  handleSteps: PropTypes.func,
  setFieldValue: PropTypes.func,
  submitForm: PropTypes.func,
  isSubmitting: PropTypes.bool,
  draftSaved: PropTypes.string,
  errors: PropTypes.any, // eslint-disable-line
  values: PropTypes.shape({
    resources: PropTypes.array, // eslint-disable-line
    resource: PropTypes.shape({
      url: PropTypes.string,
      upload: PropTypes.any, // eslint-disable-line
      name: PropTypes.string,
      description: PropTypes.string,
      mimetype: PropTypes.string,
      format: PropTypes.string,
    }),
    resourceAction: PropTypes.string,
    publish: PropTypes.bool,
    savedResources: PropTypes.number,
    lastSavedResource: PropTypes.string,
    publishing_status: PropTypes.string,
  }),
};

export default ResourceUpload;
