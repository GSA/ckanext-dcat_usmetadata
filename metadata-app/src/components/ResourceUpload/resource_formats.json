[
  {
    "label": "PPTX -- Powerpoint OOXML Presentation",
    "value": "application/vnd.openxmlformats-officedocument.presentationml.presentation"
  },
  { "label": "EXE -- Windows Executable Program", "value": "application/x-msdownload" },
  { "label": "DOC -- Word Document", "value": "application/msword" },
  { "label": "KML -- KML File", "value": "application/vnd.google-earth.kml+xml" },
  { "label": "XLS -- Excel Document", "value": "application/vnd.ms-excel" },
  { "label": "WCS -- Web Coverage Service", "value": "wcs" },
  { "label": "JS -- JavaScript", "value": "application/x-javascript" },
  { "label": "MDB -- Access Database", "value": "application/x-msaccess" },
  { "label": "NetCDF -- NetCDF File", "value": "application/netcdf" },
  { "label": "ArcGIS Map Service", "value": "ArcGIS Map Service" },
  { "label": "TSV -- Tab Separated Values File", "value": "text/tab-separated-values" },
  { "label": "WFS -- Web Feature Service", "value": null },
  { "label": "ArcGIS Online Map", "value": "ArcGIS Online Map" },
  { "label": "Perl Script", "value": "text/x-perl" },
  { "label": "KMZ -- KMZ File", "value": "application/vnd.google-earth.kmz+xml" },
  { "label": "OWL --Web Ontology Language", "value": "application/owl+xml" },
  { "label": "N3 -- N3 Triples", "value": "application/x-n3" },
  { "label": "ZIP -- Zip File", "value": "application/zip" },
  { "label": "GZ -- Gzip File", "value": "application/gzip" },
  { "label": "QGIS -- QGIS File", "value": "application/x-qgis" },
  {
    "label": "ODS -- OpenDocument Spreadsheet",
    "value": "application/vnd.oasis.opendocument.spreadsheet"
  },
  { "label": "ODT -- OpenDocument Text", "value": "application/vnd.oasis.opendocument.text" },
  { "label": "JSON -- JavaScript Object Notation", "value": "application/json" },
  { "label": "BMP -- Bitmap Image File", "value": "image/x-ms-bmp" },
  { "label": "HTML -- Web Page", "value": "text/html" },
  { "label": "RAR -- RAR Compressed File", "value": "application/rar" },
  { "label": "TIFF -- TIFF Image File", "value": "image/tiff" },
  {
    "label": "ODB -- OpenDocument Database",
    "value": "application/vnd.oasis.opendocument.database"
  },
  { "label": "TXT -- Text File", "value": "text/plain" },
  { "label": "DCR -- Adobe Shockwave format", "value": "application/x-director" },
  {
    "label": "ODF -- OpenDocument Math Formula",
    "value": "application/vnd.oasis.opendocument.formula"
  },
  { "label": "ODG -- OpenDocument Image", "value": "application/vnd.oasis.opendocument.graphics" },
  { "label": "XML -- XML File", "value": "application/xml" },
  {
    "label": "XLSX -- Excel OOXML Spreadsheet",
    "value": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
  },
  {
    "label": "DOCX -- Word OOXML Document",
    "value": "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
  },
  { "label": "BIN -- Binary Data", "value": "application/octet-stream" },
  {
    "label": "XSLT -- Extensible Stylesheet Language Transformations",
    "value": "application/xslt+xml"
  },
  { "label": "WMS -- Web Mapping Service", "value": "WMS" },
  { "label": "SVG -- SVG vector image", "value": "image/svg+xml" },
  { "label": "PPT -- Powerpoint Presentation", "value": "application/vnd.ms-powerpoint" },
  {
    "label": "ODP -- OpenDocument Presentation",
    "value": "application/vnd.oasis.opendocument.presentation"
  },
  { "label": "JPEG -- JPG Image File", "value": "image/jpeg" },
  { "label": "SPARQL -- SPARQL end-point", "value": "application/sparql-results+xml" },
  { "label": "GIF -- GIF Image File", "value": "image/gif" },
  { "label": "RDF", "value": "application/rdf+xml" },
  { "label": "E00 -- ARC/INFO interchange file format", "value": "application/x-e00" },
  { "label": "PDF -- PDF File", "value": "application/pdf" },
  { "label": "CSV -- Comma Separated Values File", "value": "text/csv" },
  { "label": "ODC -- OpenDocument Chart", "value": "application/vnd.oasis.opendocument.chart" },
  { "label": "Atom Feed -- Atom Feed", "value": "application/atom+xml" },
  { "label": "MrSID", "value": "image/x-mrsid" },
  { "label": "ArcGIS Map Preview", "value": "ArcGIS Map Preview" },
  { "label": "XYZ -- XYZ Chemical File", "value": "chemical/x-xyz" },
  { "label": "MOP -- MOPAC Input format", "value": "chemical/x-mopac-input" },
  { "label": "Esri REST -- Esri Rest API Endpoint", "value": "Esri REST" },
  { "label": "dBase -- dBase Database", "value": "application/x-dbf" },
  { "label": "MXD -- ESRI ArcGIS project file", "value": "application/x-mxd" },
  { "label": "TAR -- TAR Compressed File", "value": "application/x-tar" },
  { "label": "PNG -- PNG Image File", "value": "image/png" },
  { "label": "RSS -- RSS feed", "value": "application/rss+xml" },
  { "label": "GeoJSON -- Geographic JavaScript Object Notation", "value": null },
  { "label": "SHP -- Shapefile", "value": null }
]
