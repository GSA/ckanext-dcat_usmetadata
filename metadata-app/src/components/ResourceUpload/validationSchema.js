import * as yup from 'yup';

export default yup.object().shape({
  resource: yup.object().shape({
    name: yup.string(),
    description: yup.string(),
    mimetype: yup.string(),
    format: yup.string(),
    url: yup
      .string()
      .url(
        'If you are linking to a dataset, please include "https://" at the beginning of your URL.'
      ),
    upload: yup.mixed(),
  }),
});
