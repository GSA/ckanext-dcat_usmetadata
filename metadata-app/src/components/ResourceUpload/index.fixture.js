import React from 'react';
import { Formik } from 'formik';
import ResourceUpload from '.';
import '../../css/custom.css';

const Form = () => (
  <div className="grid-container">
    <Formik
      initialValues={{}}
      render={(values, setFieldValue) => (
        <form>
          <ResourceUpload values={values} setFieldValue={setFieldValue} />
        </form>
      )}
    />
  </div>
);

export default <Form />;
