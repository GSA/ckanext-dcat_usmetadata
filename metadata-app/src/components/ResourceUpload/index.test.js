import React from 'react';
import { Formik } from 'formik';
import { render } from '@testing-library/react';
import ResourceUpload from '.';
import '../../css/custom.css';

const Form = () => (
  <Formik initialValues={{}}>
    {(values, setFieldValue, submitForm) => (
      <form>
        <ResourceUpload values={values} setFieldValue={setFieldValue} submitForm={submitForm} />
      </form>
    )}
  </Formik>
);

test('Renders Resource Upload component', () => {
  const { container } = render(<Form />);
  expect(container).toMatchSnapshot();
});
