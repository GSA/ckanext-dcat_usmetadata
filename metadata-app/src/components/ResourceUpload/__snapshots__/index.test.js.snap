// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Resource Upload component 1`] = `
<div>
  <form>
    <div
      class="usa-form-custom"
      id="resource-upload"
    >
      <section
        class="site-component-section"
        id="section-basic-mega-menu"
      >
        <h1
          class="usite-page-title"
          id="basic-mega-menu"
        >
          Resource Upload
        </h1>
        <p
          class="site-text-intro"
        >
          You can add the URL of the dataset where it is available on the agency website. If you are uploading the dataset itself, please notify the Data.gov team at
           
          <a
            class="usa-link usa-link--external"
            href="mailto:inventory-help@gsa.gov"
            rel="noopener noreferrer"
            target="_blank"
          >
            inventory-help@gsa.gov
          </a>
          . You can also add a URL or file of information related to the dataset such as a data dictionary.
        </p>
      </section>
      
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <span
            class="usa-label"
          >
            Resource
          </span>
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="resource-option-link-to-file"
              name="resource.urlType"
              type="radio"
              value="url"
            />
            <label
              class="usa-radio__label"
              for="resource-option-link-to-file"
            >
              <b>
                Link to a file:
              </b>
               Provide a direct download link to a file
            </label>
          </div>
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="resource-option-upload-file"
              name="resource.urlType"
              type="radio"
              value="upload"
            />
            <label
              class="usa-radio__label"
              for="resource-option-upload-file"
            >
              <b>
                Upload a file:
              </b>
               Upload a file from your computer
            </label>
          </div>
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="resource-option-link-to-api"
              name="resource.urlType"
              type="radio"
              value="linkToApi"
            />
            <label
              class="usa-radio__label"
              for="resource-option-link-to-api"
            >
              <b>
                Link to an API:
              </b>
               Provide a link to access a dataset via API
            </label>
          </div>
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="resource-option-access-url"
              name="resource.urlType"
              type="radio"
              value="accessUrl"
            />
            <label
              class="usa-radio__label"
              for="resource-option-access-url"
            >
              <b>
                Access URL:
              </b>
               Provide a link to a resource that is not directly downloadable, like an html website
            </label>
          </div>
        </div>
        <div
          class="grid-col-12"
        />
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Name
            </label>
            <div
              class="usa-helptext "
            />
            
            <input
              class="usa-input  undefined undefined"
              name="resource.name"
              placeholder=""
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Description
            </label>
            <div
              class="usa-helptext "
            >
              You can use Markdown Formatting here.
            </div>
            
            <textarea
              class="usa-input  undefined undefined"
              name="resource.description"
              rows="6"
              type="string"
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Media Type
            </label>
            <div
              class="usa-helptext "
            />
            
            <select
              choices="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
              class="usa-select  "
              name="resource.mimetype"
              type="select"
            >
              <option
                value=""
              >
                -Select-
              </option>
              <option
                value="ArcGIS Map Preview"
              >
                ArcGIS Map Preview
              </option>
              <option
                value="ArcGIS Map Service"
              >
                ArcGIS Map Service
              </option>
              <option
                value="ArcGIS Online Map"
              >
                ArcGIS Online Map
              </option>
              <option
                value="application/atom+xml"
              >
                Atom Feed -- Atom Feed
              </option>
              <option
                value="application/octet-stream"
              >
                BIN -- Binary Data
              </option>
              <option
                value="image/x-ms-bmp"
              >
                BMP -- Bitmap Image File
              </option>
              <option
                value="text/csv"
              >
                CSV -- Comma Separated Values File
              </option>
              <option
                value="application/x-dbf"
              >
                dBase -- dBase Database
              </option>
              <option
                value="application/x-director"
              >
                DCR -- Adobe Shockwave format
              </option>
              <option
                value="application/msword"
              >
                DOC -- Word Document
              </option>
              <option
                value="application/vnd.openxmlformats-officedocument.wordprocessingml.document"
              >
                DOCX -- Word OOXML Document
              </option>
              <option
                value="application/x-e00"
              >
                E00 -- ARC/INFO interchange file format
              </option>
              <option
                value="Esri REST"
              >
                Esri REST -- Esri Rest API Endpoint
              </option>
              <option
                value="application/x-msdownload"
              >
                EXE -- Windows Executable Program
              </option>
              <option
                value="[object Object]"
              >
                GeoJSON -- Geographic JavaScript Object Notation
              </option>
              <option
                value="image/gif"
              >
                GIF -- GIF Image File
              </option>
              <option
                value="application/gzip"
              >
                GZ -- Gzip File
              </option>
              <option
                value="text/html"
              >
                HTML -- Web Page
              </option>
              <option
                value="image/jpeg"
              >
                JPEG -- JPG Image File
              </option>
              <option
                value="application/x-javascript"
              >
                JS -- JavaScript
              </option>
              <option
                value="application/json"
              >
                JSON -- JavaScript Object Notation
              </option>
              <option
                value="application/vnd.google-earth.kml+xml"
              >
                KML -- KML File
              </option>
              <option
                value="application/vnd.google-earth.kmz+xml"
              >
                KMZ -- KMZ File
              </option>
              <option
                value="application/x-msaccess"
              >
                MDB -- Access Database
              </option>
              <option
                value="chemical/x-mopac-input"
              >
                MOP -- MOPAC Input format
              </option>
              <option
                value="image/x-mrsid"
              >
                MrSID
              </option>
              <option
                value="application/x-mxd"
              >
                MXD -- ESRI ArcGIS project file
              </option>
              <option
                value="application/x-n3"
              >
                N3 -- N3 Triples
              </option>
              <option
                value="application/netcdf"
              >
                NetCDF -- NetCDF File
              </option>
              <option
                value="application/vnd.oasis.opendocument.database"
              >
                ODB -- OpenDocument Database
              </option>
              <option
                value="application/vnd.oasis.opendocument.chart"
              >
                ODC -- OpenDocument Chart
              </option>
              <option
                value="application/vnd.oasis.opendocument.formula"
              >
                ODF -- OpenDocument Math Formula
              </option>
              <option
                value="application/vnd.oasis.opendocument.graphics"
              >
                ODG -- OpenDocument Image
              </option>
              <option
                value="application/vnd.oasis.opendocument.presentation"
              >
                ODP -- OpenDocument Presentation
              </option>
              <option
                value="application/vnd.oasis.opendocument.spreadsheet"
              >
                ODS -- OpenDocument Spreadsheet
              </option>
              <option
                value="application/vnd.oasis.opendocument.text"
              >
                ODT -- OpenDocument Text
              </option>
              <option
                value="application/owl+xml"
              >
                OWL --Web Ontology Language
              </option>
              <option
                value="application/pdf"
              >
                PDF -- PDF File
              </option>
              <option
                value="text/x-perl"
              >
                Perl Script
              </option>
              <option
                value="image/png"
              >
                PNG -- PNG Image File
              </option>
              <option
                value="application/vnd.ms-powerpoint"
              >
                PPT -- Powerpoint Presentation
              </option>
              <option
                value="application/vnd.openxmlformats-officedocument.presentationml.presentation"
              >
                PPTX -- Powerpoint OOXML Presentation
              </option>
              <option
                value="application/x-qgis"
              >
                QGIS -- QGIS File
              </option>
              <option
                value="application/rar"
              >
                RAR -- RAR Compressed File
              </option>
              <option
                value="application/rdf+xml"
              >
                RDF
              </option>
              <option
                value="application/rss+xml"
              >
                RSS -- RSS feed
              </option>
              <option
                value="[object Object]"
              >
                SHP -- Shapefile
              </option>
              <option
                value="application/sparql-results+xml"
              >
                SPARQL -- SPARQL end-point
              </option>
              <option
                value="image/svg+xml"
              >
                SVG -- SVG vector image
              </option>
              <option
                value="application/x-tar"
              >
                TAR -- TAR Compressed File
              </option>
              <option
                value="image/tiff"
              >
                TIFF -- TIFF Image File
              </option>
              <option
                value="text/tab-separated-values"
              >
                TSV -- Tab Separated Values File
              </option>
              <option
                value="text/plain"
              >
                TXT -- Text File
              </option>
              <option
                value="wcs"
              >
                WCS -- Web Coverage Service
              </option>
              <option
                value="[object Object]"
              >
                WFS -- Web Feature Service
              </option>
              <option
                value="WMS"
              >
                WMS -- Web Mapping Service
              </option>
              <option
                value="application/vnd.ms-excel"
              >
                XLS -- Excel Document
              </option>
              <option
                value="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
              >
                XLSX -- Excel OOXML Spreadsheet
              </option>
              <option
                value="application/xml"
              >
                XML -- XML File
              </option>
              <option
                value="application/xslt+xml"
              >
                XSLT -- Extensible Stylesheet Language Transformations
              </option>
              <option
                value="chemical/x-xyz"
              >
                XYZ -- XYZ Chemical File
              </option>
              <option
                value="application/zip"
              >
                ZIP -- Zip File
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Format
            </label>
            <div
              class="usa-helptext "
            >
              Examples include: csv, xml, json.  This will be guessed automatically.  Leave blank if you wish.
            </div>
            
            <input
              class="usa-input  undefined undefined"
              name="resource.format"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <button
            class="usa-button usa-button--outline"
            type="button"
          >
            Save and add another resource
          </button>
        </div>
      </div>
      
      
      <div
        class="margin-top-6 clearfix"
      >
        <button
          class="usa-button usa-button--outline"
          type="button"
        >
          Save draft
        </button>
        <div
          class="float-right"
        >
          <button
            class="usa-button usa-button--outline"
            type="button"
          >
            Back to previous page
          </button>
          <button
            class="usa-button margin-right-0"
            type="button"
          >
            Finish and publish
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
